engine/tests/classes/Elgg/Plugins/RouteResponseTest.php

```diff
@@ -7,15 +7,12 @@
 
 /**
  * @group Router
- * @group BlogRoutes
  */
 abstract class RouteResponseTest extends IntegrationTestCase {
 
 	use PluginTesting;
 
 	public function up() {
-		$this->startPlugin();

```

Comment:
why remove this, isn't this needed to ensure the plugin is started?

Reply:
No, it was needed, because the setup wasn't working correctly

Reply:
Because we booted the app with the database, the plugin setups were loaded into the static var, but then unit tests were running without plugins, at some point integration tests start running and the setups are all messed.

Reply:
Some day, if I live till then, we will have no MySQL in core and will be able to run tests using SQLite. We can then drop the whole setup and boot the app from scratch, for now there is too much overflowing state. Fixing one things results in a cascade.

Reply:
All i see is that the parent checks if this test is run from a plugin and skips it if the plugin isn't active. Am i missing something obvious?

Reply:
Or do we rely on https://github.com/Elgg/Elgg/blob/master/.scripts/travis/enable_plugins.php

Reply:
`start.php` is required and added `$_setups` in `Application`.  Next time application is boot, it doesn't check whether the plugin is loaded, and just runs the setup, which registers the `init` handler for a plugin that's neither loaded nor booted (in case of unit tests).

Reply:
https://github.com/Elgg/Elgg/issues/11654 would make all this better, correct?

Reply:
`bootstrap.php` initializes the application with all plugins that are active, populates the cache, and from there is all goes to shambles, because plugin boot is a convoluted process that's neither transparent nor comprehensible. Plugins are not isolated enough to be booted with the application on multiple consecutive boots. `ElggPlugin::start` is a painful mess that runs at runtime, so it ignores all system events. Basically to make plugins testable we need to throw away legacy `start.php` and either wrap required files into an event handler, or use classes for boot and init.

Reply:
> #11654 would make all this better, correct?

Only if there 2 bootstraps for each suite - one with db and one without

Reply:
Unless the `Application` is decouple from core loading, then we can just load core files without booting the application.